libs/wingsdk/src/cloud/stream.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Stream
+id: stream
+description: A built-in resource for working with streaming data endpoints.
+keywords:
+  [
+    Wing reference,
+    Wing language,
+    language,
+    Wing standard library,
+    Wing programming language,
+    Streaming Data,
+  ]
+sidebar_position: 1
+---
+
+The `cloud.Stream` resource represents a real-time streaming service typically used to ingest continuous data transmissions.
+They typically involve high throughput, low size transmissions like: sensors, mobile devices, web servers, etc.
+
+When a data stream is created in a cloud provider, it typically exposes an endpoint acting as a "put" endpoint for all data sources. The cloud provider manages the ingest and temporary storage of your data. 
+
+Most cloud providers have a data horizon for their streaming data to help providers reduce cost, prevent unbounded data growth, encourage real-time processing, and implement data governance best practices. However, developers have the ability to modify the data horizon.
+
+## Usage
+
+```ts
+bring cloud;
+
+let stream = new cloud.Stream(
+    name: 'test-data-stream',
+    horizon: 96,  // optional, defaults to 24 hours
+);
+```
+
+## Advanced Usage
+
+If the expected throughput of your data is static and known, you can leverage provisioned throughput.
+
+```ts
+bring cloud;
+
+let stream = new cloud.Stream(
+  name: ,
```

Comment:
The name can probably be omitted -- each preflight class has a construct ID that can be used to customize for the resource's name. The syntax might change in the future but right now it's customizable like this:

```ts
let stream = new cloud.Stream(
   ...options
) as "MyStream";
```